flow-server/src/test/java/com/vaadin/flow/component/SvgTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SvgTest {
+
+    @Test
+    public void attachedToElement() {
+        // This will throw an assertion error if the element is not attached to
+        // the component
+        new Svg("<svg></svg>").getParent();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void nullSvg() {
+        new Svg((String) null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void nullStream() {
+        new Svg((InputStream) null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void emptySvg() {
+        new Svg("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void text() {
+        new Svg("hello");
+    }
+
+    static String TRIVIAL_SVG = """
+            <svg>
+                <circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow" />
+            </svg>
+                                    """;
```

Comment:
(Not really that important)
Is the extra line and empty string in the end intentional? Otherwise, we can remove it:
```suggestion
            </svg>""";
```

Reply:
Same for the next SVG static content definitions.